### PR TITLE
Improve Issue template

### PR DESCRIPTION
**Pull request by sesam**
Improve Issue templates

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [X] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Sorry, I don't agree with these edits. Can you elaborate on your motivation for them?

**Comment by MikeMcQuaid** [comment]
Can you explain what's the main change you're hoping to make to these templates and why?

**Comment by sesam** [comment]
tl;dr? :)

- all instructions on top
- more important things before less
- simplify, preferably even more

But I expected some resistance :) and didn't have time to make the diff more awesome.

The first thing that hit me was the last thing in the ISSUE TEMPLATE, which I hadn't read. I had assumed the template followed the convention of all such templates and official forms I'd seen before; instructions on top, fill-in things below. Feeling that "Feature request" was kind of off-topic - isn't everybody going to ugprade to High Sierra anyways? - and I read 10% of the words of that section. I can't remember ever reading the last line before getting my issues insta-closed. I was quite taken aback when I got the answer that you're not planning to spend more than a few seconds pondering whether to help High Sierra-casualties avoid hard to debug issues.

It was just too much text at the time. Maybe to native English speakers or real programmers, or those who don't work late at night, this never happens. On the other hand, do you have any metrics on how efficiently you attract contributors vs other projects? Maybe hard to measure.

When I came into this, I was in a hurry to deliver a project, and got stuck 3 hours figuring a lot of things out, basically since I had already missed my deadline and I didn't want that to happen to the next guy. I didn't take time to carefully and methodically go through the ISSUE TEMPLATE text. So ideally I'd like to shorten and simplify the bug reporting process even further. Move more of the troubleshooting and triage work closer to the user; I assume you get way too much "support requests" coming in as issues, and it seems the two of you are doing the lions share as-is.
If I had even more time, I'd wish to make a more context sensitive, guided approach to detecting and containing bugs and troubleshooting. Maybe a `brew trouble` that self-advertises whenever there's trouble brewing in your /usr/local folders. Also I see there's an autocomplete feature that's not installed or suggested by the default installation. But due to time constraints, I had to leave it at this.

I have one unfinished stash for the ISSUE_TEMPLATE.md file; better paste it before I go away:
````
-# Features
-Please replace this section with:
+# Adding a feature?
+For software requests [go here](PASTE_URL_HERE)
+Then fill in below:
````

**Comment by scpeters** [comment]
The one change I like is adding `Thanks for making Homebrew better!` in the top block, which makes it a little less scary.

For me, people not reading the issue template isn't a good reason to move the last two lines to the top.

Also, you delete the reference to `homebrew-core`, which is where formula-related issues should be posted.

**Comment by sesam** [comment]
I believe in making it hard to make mistakes.
Optimally there'd be a single webpage, say brew.sh/help where people would be guided through the process of figuring out what their problem is, and if it can be down to a bug, or a feature request, and managing expectations. Maybe deflect as much support-type questions as possible onto stackexchange-sites, giving people a shared learning experience there instead of an uphill struggle here :)

While such a joint "triage help" page doesn't exist, I picked the next best thing, and (after carefullly looking at the linked homebrew-core page) replaced it with a link to the troubleshooter for homebrew-core.
That might deflect some who'd otherwise make an issue on "brew" simply because it looked legit and they didn't take half an hour to get properly accustomed to how this project has structured things. Maybe I'm assuming too much in thinking I'm not the only one who at first glance thinks that "brew" means all of brew, even when I've already cloned the Formula-repo and have a formula "cooking". I just never remember repo names that aren't named very much the same as what I expect them to contain.
If you ask me in 2-3 months time if "brew" might be where bugs with formulas are to be reported, I'd only know to answer that there's something tricky with brew - careful where you post things - it's not too easy to nagivate.

**Comment by MikeMcQuaid** [comment]
> tl;dr? :)
> didn't have time
> which I hadn't read
> just too much text
> due to time constraints

I'm the lead maintainer of Homebrew (and various other projects), an engineer at GitHub, a parent to a young baby, training for a sports competition, a husband and dog owner. Diplomatically, repeatedly referencing that you are too busy to read the instructions requires you to prioritise how you spend your time better rather than optimise our instructions for people who don't want to read them.

> do you have any metrics on how efficiently you attract contributors vs other projects?

We have some of the most active and large numbers of contributors for any volunteer run GitHub project.

> The one change I like is adding Thanks for making Homebrew better! in the top block, which makes it a little less scary.

I agree it makes it less scary but filing a bad issue should be a little scary; if you don't file it well you are making Homebrew worse rather than better.

> For me, people not reading the issue template isn't a good reason to move the last two lines to the top.

A good summary. Closing, sorry.

**Comment by sesam** [comment]
I didn't mean to offend, just tease a bit.

**Comment by MikeMcQuaid** [comment]
@sesam No offence taken, don't worry, just pointing out that it's generally not a good line to take with people 😄.
